src/zarr/api/asynchronous.py

```diff
@@ -109,10 +109,10 @@ def _get_shape_chunks(a: ArrayLike | Any) -> tuple[ChunkCoords | None, ChunkCoor
     return shape, chunks
 
 
-def _like_args(a: ArrayLike, kwargs: dict[str, Any]) -> dict[str, Any]:
+def _like_args(a: ArrayLike) -> dict[str, object]:
```

Comment:
Why the change to `object`? And if you're feeling eager, this could maybe be a typed dict?

```
class LikeArgs(TypedDict):
    shape: tuple[int, ...]
    chunks: tuple[int, ...]
    dtype: np.typing.DTypeLike
    order: ...
```


but that's probably out of scope for this PR.

Reply:
and re: the typeddict thing, yeah that would be nice. typeddictification across the codebase is on my to-do list.